blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricIO.java

```diff
@@ -206,7 +292,7 @@ public ResultSetFuture putAsync(Locator locator, long collectionTime, Rollup rol
      * @param rollup
      * @return
      */
-    protected abstract <T extends Rollup> ByteBuffer toByteBuffer(T rollup);
+    protected abstract <T extends Object> ByteBuffer toByteBuffer(T rollup);
```

Comment:
Parameter name should be something else if it's being used for things other than rollup.